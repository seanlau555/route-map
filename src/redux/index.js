export { rootReducer } from './rootReducer'
export { rootSaga } from './rootSaga'

export { appActionCreators } from './app/actions'
