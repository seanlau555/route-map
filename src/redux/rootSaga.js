import { all, fork } from 'redux-saga/effects'
import { appSaga } from './app/saga'

export const combineSagas = sagas =>
  function* rootSaga() {
    try {
      yield all(sagas.map(fork))
    } catch (err) {
      console.error(err)
    }
  }

export const rootSaga = combineSagas([appSaga])
