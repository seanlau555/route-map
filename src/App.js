import React from 'react'
import { Row, Col } from 'antd'
import { connect } from 'react-redux'
import './App.css'
import { appActionCreators } from './redux'
import Inputs from './components/Inputs'
import Map from './components/Map'
import { routeSelector } from './redux/app/selectors'

class App extends React.PureComponent {
  render() {
    const {
      route,
      isFetching,
      PostRoute,
      origin,
      destination,
      errorMessage
    } = this.props
    return (
      <div className="App">
        <Row gutter={16}>
          <Col xs={{ span: 24 }} md={{ span: 8 }}>
            <Inputs
              route={route}
              errorMessage={errorMessage}
              PostRoute={PostRoute}
            />
          </Col>
          <Col xs={{ span: 24 }} md={{ span: 16 }}>
            <Map
              origin={origin}
              destination={destination}
              isFetching={isFetching}
              route={route}
            />
          </Col>
        </Row>
      </div>
    )
  }
}

const mapStateToProps = ({ app }) => ({
  token: app.token,
  route: routeSelector({ app }),
  isFetching: app.isFetching,

  origin: app.origin,
  destination: app.destination,
  errorMessage: app.errorMessage
})

const mapDispatchToProps = {
  PostRoute: appActionCreators.postRoute
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
