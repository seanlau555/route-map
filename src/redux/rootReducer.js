import { combineReducers } from 'redux'

import { appReducer } from './app/reducer'

export const rootReducer = combineReducers({
  app: appReducer
})
