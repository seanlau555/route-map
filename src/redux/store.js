import {
  applyMiddleware,
  compose,
  createStore as createReduxStore
} from 'redux'
import createSagaMiddleware from 'redux-saga'
import { rootReducer } from './rootReducer'
import { rootSaga } from './rootSaga'

export function createStore(initialState) {
  const sagaMiddleware = createSagaMiddleware()

  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

  const store = createReduxStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(sagaMiddleware))
  )

  sagaMiddleware.run(rootSaga)

  return store
}
