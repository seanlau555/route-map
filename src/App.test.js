import React from 'react'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import { createStore } from './redux/store'

import App from './App'

describe('AppContainer', () => {
  it('should connect store correctly', () => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn()
      }))
    })
    const props = {
      token: '',
      route: {},
      isFetching: false,
      origin: '',
      destination: '',
      errorMessage: '',
      PostRoute: jest.fn()
    }
    const wrapper = mount(
      <Provider store={createStore()}>
        <App  {...props} />
      </Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
