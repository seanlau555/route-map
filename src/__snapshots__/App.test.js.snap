// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer should connect store correctly 1`] = `ReactWrapper {}`;
